Packages/com.llealloo.audiolink/Runtime/Scripts/AudioLink.cs

```diff
@@ -545,10 +561,21 @@ private void Update()
 
                 // Used to correct for the volume of the audio source component
 
-                audioMaterial.SetFloat(_SourceVolume, audioSource.volume);
-                audioMaterial.SetFloat(_SourceSpatialBlend, audioSource.spatialBlend);
-                audioMaterial.SetVector(_SourcePosition, audioSource.transform.position);
+                float sourceVolume = audioSource.volume;
+                Vector3 sourcePosition = audioSource.transform.position;
+                float sourceMax = audioSource.maxDistance;
 
+                AnimationCurve sourceBlendCurve = audioSource.GetCustomCurve(AudioSourceCurveType.SpatialBlend);
+                AnimationCurve sourceFalloffCurve = audioSource.GetCustomCurve(AudioSourceCurveType.CustomRolloff);
+                var playerEars = GetHearingLocation();
+                float listeningDistance = Vector3.Distance(playerEars, sourcePosition) / sourceMax;
+                float sourceFalloff = sourceFalloffCurve.Evaluate(listeningDistance);
+                float sourceBlend = sourceBlendCurve.Evaluate(listeningDistance);
+                sourceVolume = Mathf.Lerp(sourceVolume, sourceVolume * sourceFalloff, sourceBlend);
```

Comment:
I'm not too familiar with this API, so just curious: Are you mimicking behavior that is specified somewhere? Since we were just using audioSource.volume before. Or is this empirical?